app/assets/javascripts/modules/single-file-upload.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+moj.Modules.singleFileUpload = {
+  input_class: '.single-file-upload',
+  spinnerImagePath: '/assets/spinner.gif',
+
+  init: function() {
+    var self = this;
+
+    if($(self.input_class).length) {
+      self.bindEvents();
+      self.preloadSpinner();
+    }
+  },
+
+  bindEvents: function() {
+    var self = this,
+        $el = $(self.input_class).eq(0),
+        $form = $el.closest('form');
+
+    $el.on('change', function() {
+      var statusText = '';
+
+      if($el.val()) {
+        statusText = 'SELECTED';
```

Comment:
ISTR it was more about the order in which things were being evaluated. When the JS was loaded, the relevant rails config value was undefined.
I'm sure we could find a way to make it work, but for the sake of getting this done (rather than diving down a rabbit hole) I'd suggest we go with this as is, and write a ticket in the icebox for moving the text into a translation file.